clients/go/cacclient/main.go

```diff
@@ -0,0 +1,97 @@
+package cacclient
+
+/*
+#include "libcac_client.h"
+*/
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+// CacClient struct
+type CacClient struct {
+	tenant           string
+	pollingFrequency int
+	cacHostName      string
+}
+
+// NewCacClient creates a new CacClient
+func NewCacClient(tenantName string, pollingFrequency int, cacHostName string) *CacClient {
+	return &CacClient{tenant: tenantName, pollingFrequency: pollingFrequency, cacHostName: cacHostName}
+}
+
+// GetCacLastErrorMessage gets the last error message from the C library
+func (c *CacClient) GetCacLastErrorMessage() string {
+	return C.GoString(C.cac_last_error_message())
+}
+
+// GetCacLastErrorLength gets the length of the last error message
+func (c *CacClient) GetCacLastErrorLength() int {
+	return int(C.cac_last_error_length())
+}
+
+// GetCacClient gets the client pointer from the C library
+func (c *CacClient) GetCacClient() *C.Arc_Client {
+	return C.cac_get_client(C.CString(c.tenant))
+}
+
+// CreateNewCacClient creates a new client in the C library
+func (c *CacClient) CreateNewCacClient() int {
+	resp := C.cac_new_client(C.CString(c.tenant), C.ulong(c.pollingFrequency), C.CString(c.cacHostName))
+	if resp == 1 {
+		errorMessage := c.GetCacLastErrorMessage()
+		fmt.Printf("Some Error Occur while creating new client: %s\n", errorMessage)
+	}
+	return int(resp)
+}
+
+// StartCacPollingUpdate starts polling updates in a separate goroutine
+func (c *CacClient) StartCacPollingUpdate() {
+	tenant := C.CString(c.tenant)
+	go C.cac_start_polling_update(tenant)
+}
+
+// GetCacConfig gets the configuration from the C library
+func (c *CacClient) GetCacConfig(filterQuery string, filterPrefix string) string {
```

Comment:
params like filter query should take Maybe Json object and context should take Json object type of data, and then the function should stringify the json value and pass it to the FFI functions

in case of no json, simply pass null pointer, 

it needs to be the same across other clients as well, please make the changes there as well
and if possible raise a fix PR for python client as well, adding the same changes

Reply:
@ayushjain17 updated